common/db/buffered_bulk.go

```diff
@@ -72,6 +76,34 @@ func (bb *BufferedBulkInserter) Insert(doc interface{}) error {
 	return err
 }
 
+// FlushWithRetry continously writes all buffered documents in one bulk insert until there's no error then resets the buffer.
+func (bb *BufferedBulkInserter) FlushWithRetry() error {
+	failedInsertCount := 0
+	if bb.docCount == 0 {
+		return nil
+	}
+	defer bb.resetBulk()
+retry:
+	if _, err := bb.bulk.Run(); err != nil {
+		if strings.Contains(err.Error(), "Request rate is large") ||
```

Comment:
Does the exception contain any indication of the block partially succeeding, or the suggested retry wait?

Reply:
When I attempt to print out the error object I just gives a string
````
Message: {"Errors": ["Request rate is large"]}
ActivityId: --omitted--
Request URI: --omitted--
Request status: --empty--
SDK: Microsoft.Azure.Documents.Common/2.0.0.0

Reply:
Interestingly enough, the [.Run()](https://github.com/SimonJinaphant/mongo-tools/blob/cc40051150464474d3fb8425beea62e4f0652ca9/vendor/src/gopkg.in/mgo.v2/bulk.go#L274) method for mgo seems to return a struct call `BulkResult` (the `_` variable in the above code) which is documented to hold the result for a bulk operation but it seems [barely implemented](https://github.com/SimonJinaphant/mongo-tools/blob/master/vendor/src/gopkg.in/mgo.v2/bulk.go#L50)